exporter.go

```diff
@@ -709,16 +709,21 @@ func (e *Exporter) handleMetricsCommandStats(ch chan<- prometheus.Metric, fieldK
 	e.registerConstMetric(ch, "commands_duration_seconds_total", usecTotal/1e6, prometheus.CounterValue, cmd)
 }
 
-func (e *Exporter) handleMetricsReplication(ch chan<- prometheus.Metric, fieldKey string, fieldValue string) bool {
+func (e *Exporter) handleMetricsReplication(ch chan<- prometheus.Metric, masterHost string, masterPort string, fieldKey string, fieldValue string) bool {
 	// only slaves have this field
 	if fieldKey == "master_link_status" {
 		if fieldValue == "up" {
-			e.registerConstMetricGauge(ch, "master_link_up", 1)
+			e.registerConstMetricGauge(ch, "master_link_up", 1, masterHost, masterPort)
 		} else {
-			e.registerConstMetricGauge(ch, "master_link_up", 0)
+			e.registerConstMetricGauge(ch, "master_link_up", 0, masterHost, masterPort)
 		}
 		return true
 	}
+	if fieldKey == "master_last_io_seconds_ago" || fieldKey == "slave_repl_offset" || fieldKey == "master_sync_in_progress" {
```

Comment:
Can you redo this if statement with the one from line 714 and make it a switch on fieldKey?

Reply:
I can change to switch